src/vue/mixins/asset-loader.mixin.js

```diff
@@ -0,0 +1,28 @@
+import { Sdk } from '@/sdk'
+import { ErrorHandler } from '@/js/helpers/error-handler'
+
+import { vuexTypes } from '@/vuex'
+import { mapGetters } from 'vuex'
+
+export default {
+  data: _ => ({
+    ownedAssets: []
+  }),
+  computed: {
+    ...mapGetters([
+      vuexTypes.account
+    ])
+  },
+  methods: {
+    async loadOwnedAssets () {
+      try {
+        const response = await Sdk.horizon.assets.getAll()
+        this.ownedAssets = response.data
```

Comment:
making the request to the `/assets?owner=YOURACCOUNTID` will reduce the response a lot

Reply:
and you won't need to filter the result

Reply:
and can remove this method at all, just directly querying sdk

Reply:
Resolved, updated

Reply:
I cannot remove this method, cause it's called many times outside